pkg/util/iptables/iptables.go

```diff
@@ -524,6 +481,73 @@ func (runner *runner) checkRuleUsingCheck(args []string) (bool, error) {
 	return false, fmt.Errorf("error checking rule: %v: %s", err, out)
 }
 
+const (
+	// Max time we wait for an iptables flush to complete after we notice it has started
+	iptablesFlushTimeout = 5 * time.Second
+	// How often we poll while waiting for an iptables flush to complete
+	iptablesFlushPollTime = 100 * time.Millisecond
+)
+
+// Monitor is part of Interface
+func (runner *runner) Monitor(reloadFunc func(), interval time.Duration, stopCh <-chan struct{}) {
+	bytes := make([]byte, 5)
+	rand.Read(bytes)
```

Comment:
Yes, it will leak chains, but I'm not sure why you think it will force a reload... ?

The leak is unfixable the way things are now, since we can't tell the difference between a chain that was leaked by a previous run of the same agent vs a chain that is still in use by another process. So I guess that's an argument for caller-passes-in-chain-name!